notebooks/MAST/astroqueryMAST/beginner_program_id_search/companion_script.py

```diff
@@ -0,0 +1,48 @@
+import sys
+from astroquery.mast import Observations
+
+def download_by_jwst_pid (pid, curl_flag):
+    """
+    This function retrieves all of the level 2 science products for a single program ID.
+    Inputs:
+    pid [int] – program ID number
+    curl_flag [Boolean] – If set to true, function returns a curl script.
+    """
+
+    # Login, search for observations
+    Observations.login(store_token=True)
+    obs_list = Observations.query_criteria(proposal_id=pid)
+    data_products = Observations.get_product_list(obs_list)
+    filtered_prod = Observations.filter_products(data_products, calib_level=[2], productType='SCIENCE')
+
+    # Filesize check
```

Comment:
Can use same re-write for total size as mentioned in the Python notebook for improved readability.